### PR TITLE
Currys

### LINKED ISSUE #2840: Spider currys is broken
During the global build at 2021-09-08-14-42-47, spider **currys** failed with **0 features** and **0 errors**.

Here's [the log](https://data.alltheplaces.xyz/runs/2021-09-08-14-42-47/logs/currys.txt) and [the output](https://data.alltheplaces.xyz/runs/2021-09-08-14-42-47/output/currys.geojson) ([on a map](https://data.alltheplaces.xyz/map.html?show=https://data.alltheplaces.xyz/runs/2021-09-08-14-42-47/output/currys.geojson))

### PR DESCRIPTION
**Pull request by Cj-Malone**
I can't figure out why I'm getting a 403 on this, same with curl.

Once the 403 is sorted, this should fix #2840

### CODE REVIEW

**Comment by Cj-Malone** [comment]
This relates to some Cloudflare stuff, they are rejecting it as bot activity based on the TLS handshake.

File: locations/spiders/currys.py [hunk 1]
```diff
@@ -1,60 +1,39 @@
...
-        search_url = "https://www.pcworld.co.uk/gbuk/s/find-a-store.html"
-        for country in countries:
-            formdata["sStoreKeyword"] = country
-            request = scrapy.FormRequest(
-                search_url, formdata=formdata, callback=self.parse
-            )
-            request.meta["country"] = country
-            yield request
+    allowed_domains = ["www.currys.co.uk"]
+    sitemap_urls = ["https://www.currys.co.uk/sitemap-stores.xml"]
```
Comment (UKChris-osm) [inline locations/spiders/currys.py:13]: Not sure where you're at with the Currys bot, but the URL for their stores XML file is https://www.currys.co.uk/sitemap-stores-curryspcworlduk.xml as opposed to the one listed in this PR - hope that helps.

Comment (Cj-Malone) [reply locations/spiders/currys.py]: Thanks, I've updated this, but the issue here is with @cloudflare they are blocking the traffic. You can test with: `$ curl https://www.currys.co.uk/sitemap-stores-curryspcworlduk.xml`

**Comment by Cj-Malone** [comment]
@iandees I think there are 2 options right now:

1. We either include it with `requires_proxy` and hopefully solve it at the same time, although they aren't exactly the same issue.
2. We add playwright as a dependency for this spider/other Cloudflare ones.

**Comment by iandees** [comment]
I think we should do both. Let's try playwright and see what happens. It'd be interesting to figure out how to do playwright on Github Actions and in the Docker container.

**Comment by Cj-Malone** [comment]
[log](https://data.alltheplaces.xyz/ci/4036387043/currys/log.txt) [map](https://data.alltheplaces.xyz/map.html?show=https://data.alltheplaces.xyz/ci/4036387043/currys/output.geojson)

I think the errors are just about [bad urls](https://www.currys.co.uk/page?cid=leeds-crown-point-2267) in the sitemap, not actual errors.

I definitely recommend testing Docker before the next weekly.

```python
{'atp/brand/currys': 293,
 'atp/brand_wikidata/Q3246464': 293,
 'atp/category/shop/electronics': 293,
 'atp/field/city/missing': 74,
 'atp/field/email/missing': 293,
 'atp/field/image/missing': 293,
 'atp/field/phone/missing': 293,
 'atp/field/state/missing': 293,
 'atp/field/twitter/missing': 293,
 'atp/nsi/perfect_match': 293,
 'downloader/exception_count': 11,
 'downloader/exception_type_count/playwright._impl._api_types.Error': 11,
 'downloader/request_bytes': 185591,
 'downloader/request_count': 308,
 'downloader/request_method_count/GET': 308,
 'downloader/response_bytes': 293637382,
 'downloader/response_count': 297,
 'downloader/response_status_count/200': 294,
 'downloader/response_status_count/503': 3,
 'elapsed_time_seconds': 385.608989,
 'feedexport/success_count/FileFeedStorage': 1,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2023, 1, 29, 9, 54, 18, 337798),
 'httpcache/firsthand': 295,
 'httpcache/hit': 2,
 'httpcache/miss': 306,
 'httpcache/store': 295,
 'item_scraped_count': 293,
 'log_count/DEBUG': 11258,
 'log_count/ERROR': 12,
 'log_count/INFO': 18,
 'log_count/WARNING': 11,
 'memusage/max': 724557824,
 'memusage/startup': 116723712,
 'playwright/context_count': 1,
 'playwright/context_count/max_concurrent': 1,
 'playwright/context_count/non-persistent': 1,
 'playwright/page_count': 305,
 'playwright/page_count/closed': 305,
 'playwright/page_count/max_concurrent': 3,
 'playwright/request_count': 10027,
 'playwright/request_count/aborted': 9733,
 'playwright/request_count/method/GET': 9583,
 'playwright/request_count/method/POST': 444,
 'playwright/request_count/navigation': 305,
 'playwright/request_count/resource_type/document': 305,
 'playwright/request_count/resource_type/image': 2590,
 'playwright/request_count/resource_type/other': 444,
 'playwright/request_count/resource_type/script': 5516,
 'playwright/request_count/resource_type/stylesheet': 1172,
 'playwright/response_count': 294,
 'playwright/response_count/method/GET': 294,
 'playwright/response_count/resource_type/document': 294,
 'request_depth_max': 1,
 'response_received_count': 295,
 'retry/count': 2,
 'retry/max_reached': 1,
 'retry/reason_count/503 Service Unavailable': 2,
 'robotstxt/request_count': 1,
 'robotstxt/response_count': 1,
 'robotstxt/response_status_count/503': 1,
 'scheduler/dequeued': 305,
 'scheduler/dequeued/memory': 305,
 'scheduler/enqueued': 305,
 'scheduler/enqueued/memory': 305,
 'start_time': datetime.datetime(2023, 1, 29, 9, 47, 52, 728809)}
```

**Comment by iandees** [comment]
The Dockerfile build seems to be broken (at least locally) on this branch, so I'm looking at how to fix it still.

**Comment by Cj-Malone** [comment]
CI is now failing because it thinks there is more than 5 spiders changed. Other than the docker thing, I think this is good to go.

**Pull request merged**